builtin/providers/aws/resource_aws_emr_cluster.go

```diff
@@ -225,6 +248,46 @@ func resourceAwsEMRClusterCreate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if a, ok := d.GetOk("ebs_volume"); ok {
+		instanceConfigEbs := &instanceConfig
+		ebsVolume := a.([]interface{})
+		volume := ebsVolume[0].(map[string]interface{})
+
+		sizeInGB := volume["size"].(int)
+
+		volumeType := volume["type"].(string)
+
+		volumesPerInstance := volume["volumes_per_instance"].(int)
+
+		configEbs := []*emr.InstanceGroupConfig{
+			{
+				InstanceCount: aws.Int64(int64(coreInstanceCount)),
+				InstanceRole:  aws.String(d.Get("service_role").(string)),
+				InstanceType:  aws.String(coreInstanceType),
+				EbsConfiguration: &emr.EbsConfiguration{
+					EbsBlockDeviceConfigs: []*emr.EbsBlockDeviceConfig{
+						{
+							VolumeSpecification: &emr.VolumeSpecification{
+								SizeInGB:   aws.Int64(int64(sizeInGB)),
+								VolumeType: aws.String(volumeType),
+							},
+							VolumesPerInstance: aws.Int64(int64(volumesPerInstance)),
+						},
+					},
+				},
+			},
+		}
+
+		*instanceConfigEbs = &emr.JobFlowInstancesConfig{
+			MasterInstanceType:          aws.String(masterInstanceType),
+			SlaveInstanceType:           aws.String(coreInstanceType),
+			InstanceCount:               aws.Int64(int64(coreInstanceCount)),
+			KeepJobFlowAliveWhenNoSteps: aws.Bool(true),
+			TerminationProtected:        aws.Bool(false),
```

Comment:
I think KeepJobFlowAliveWhenNoSteps wouldn't be used for Terraform managed clusters.

Giving an option to change TerminationProtected on/off would be great though.